frontend/packages/dev-console/src/components/pipelineruns/events/event-utils.ts

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import {
+  WatchK8sResources,
+  useK8sWatchResources,
+  WatchK8sResults,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { PodModel } from '@console/internal/models';
+import {
+  EventInvolvedObject,
+  K8sResourceKind,
+  referenceForModel,
+} from '@console/internal/module/k8s';
+import { PipelineRun, TaskRunKind } from '../../../utils/pipeline-augment';
+import { PipelineRunModel, TaskRunModel } from '../../../models';
+import { TektonResourceLabel } from '../../pipelines/const';
+
+export type EventFilter = (eventObj: EventInvolvedObject) => boolean;
+type ResourcesType = { [key: string]: K8sResourceKind[] };
+/**
+ * Returns the pod resource for watchK8sResource
+ * @param namespace
+ * @param labels metadata lables to query
+ */
+export const getPodsByLabels = (namespace: string, labels: { [label: string]: string }) => {
+  return {
+    kind: PodModel.kind,
+    selector: {
+      matchLabels: labels,
+    },
+    namespace,
+    isList: true,
+    optional: true,
+  };
+};
+/**
+ * Get associated taskruns and pods for a given namespace.
+ * @param namespace
+ */
+export const usePipelineRunRelatedResources = (
+  namespace: string,
+  pipelineRunName: string,
+): WatchK8sResults<ResourcesType> => {
+  const plrRelatedResources: WatchK8sResources<ResourcesType> = React.useMemo(() => {
+    return {
+      taskruns: {
+        kind: referenceForModel(TaskRunModel),
+        namespace,
+        selector: {
+          matchLabels: { [TektonResourceLabel.pipelinerun]: pipelineRunName },
+        },
+        isList: true,
+        optional: true,
+      },
+      pods: getPodsByLabels(namespace, { [TektonResourceLabel.pipelinerun]: pipelineRunName }),
+    };
+  }, [namespace, pipelineRunName]);
+  return useK8sWatchResources<ResourcesType>(plrRelatedResources);
+};
```

Comment:
instead of having 2 separate hooks `usePipelineRunRelatedResources` , `usePipelineRunFilters` can we have 

```
const plrRelatedResources: WatchK8sResources<ResourcesType> = React.useMemo(() => {
    return {
      taskruns: {
        kind: referenceForModel(TaskRunModel),
        namespace,
        selector: {
          matchLabels: { [TektonResourceLabel.pipelinerun]: pipelineRunName },
        },
        isList: true,
        optional: true,
      },
      pods: getPodsByLabels(namespace, { [TektonResourceLabel.pipelinerun]: pipelineRunName }),
    };
  }, [namespace, pipelineRunName])
```

and use `useK8sWatchResources<ResourcesType>(plrRelatedResources);` in hook `usePipelineRunFilters`

Reply:
The Reason  I decoupled this `usePipelineRunRelatedResources` hook from `usePipelineRunFilters` is, the `usePipelineRunFilters` returns an array filter functions specific to the `ResourcesEventStream` component. So, later if we need the `taskRuns` and `Pods` associated to a pipelineRun for some other usecase, then we call this common hook `usePipelineRunRelatedResources`.